src/core/EdFi.DataManagementService.Core/Response/FailureResponse.cs

```diff
@@ -92,4 +93,29 @@ public static BaseFailureResponse ForInvalidReferences(ResourceName[] resourceNa
             correlationId: null
         );
     }
+
+    public static FailureResponseWithErrors ForImmutableIdentity(ResourceName[] resourceNames) =>
+        new(
+            detail:
+            $"The referenced {string.Join(",", resourceNames.Select(x => x.Value))} item(s) do not exist.",
+            type: $"{_dataConflictTypePrefix}:unresolved-reference",
+            title: "Unresolved Reference",
+            status: 409,
+            correlationId: null,
+            validationErrors: null,
+            errors: null
+        );
+
+    public static FailureResponseWithErrors ForKeyChangeNotSupported(string error) =>
```

Comment:
Rename this method to ForImmutableIdentity. Really. The end user message "Key Change Not Supported" is correct but irrelevant to us. We don't have an internal concept of "key changes". We have "identity changes".  "ImmutableIdentity" means the identity cannot be changed.